docs/src/index.md

```diff
@@ -1,14 +1,185 @@
+Conversion to `RGB{Float32}` may be safer than `RGB{T}` where `T` is limited to 0-to-1.
+It is also faster, as the result does not have to be checked for whether it exceeds the bounds of the type.
+(To prevent overflow, all internal operations are performed using floating-point intermediates even if you want a `FixedPoint` output.)
+
+!!! note
+    While `ColorMixture` objects can be converted to RGB, they are *not* AbstractRGB
+    colors: `red(c)`, `green(c)`, and `blue(c)` are not defined for `c::ColorMixture`, and low-level utilities
+    like `mapc` operate on the raw channel intensities rather than the RGB values.
+
+
+There are several ways you can create these colors. An easy approach is to define the type through a "template" object:
+
+```jldoctest demo
+julia> ctemplate = ColorMixture{Float32}((RGB(0,1,0), RGB(1,0,0)))
+(0.0₁, 0.0₂)
 ```
 
-```@autodocs
-Modules = [FluorophoreColors]
+`ctemplate` is an all-zeros `ColorMixture` object, but can be used to construct arbitrary `c` with specified intensities:
+
+```jldoctest demo
+julia> typeof(ctemplate)
+ColorMixture{Float32, 2, (RGB{N0f8}(0.0,1.0,0.0), RGB{N0f8}(1.0,0.0,0.0))}
+
+julia> c = ctemplate(0.2, 0.4)
+(0.2₁, 0.4₂)
+
+julia> Tuple(c)
+(0.2f0, 0.4f0)
+```
+
+You can also create them with a single call `ColorMixture(rgbs, intensities)`:
+
+```jldoctest demo
+julia> c = ColorMixture{Float32}((RGB(0,1,0), RGB(1,0,0)), (0.2, 0.4))
+(0.2₁, 0.4₂)
+```
+
+or even by explicit type construction:
+
+```jldoctest demo
+julia> ColorMixture{Float32, 2, (RGB{N0f8}(0.0,1.0,0.0), RGB{N0f8}(1.0,0.0,0.0))}(0.2, 0.4)
+(0.2₁, 0.4₂)
 ```
+
+!!! tip
+    All but the last form require [constant propagation](https://en.wikipedia.org/wiki/Constant_folding) for inferrability.
+    Julia 1.7 and higher can use "aggressive" constant propagation to solve inference problems that may reduce performance on Julia 1.6.
+
+### Importing external data
+
+When objects are not created by code but instead loaded from an external source such as a file, you have several avenues for creating arrays of multichannel color objects. There are two particularly common cases:
+
+1. If the imported data are an array `A` of size `(nc, m, n)`, where `nc` is the number of color channels (i.e., color is the fastest dimension), then use `reinterpret(reshape, C, A)` where `C` is the color type you want to use (e.g., `MultiChannelColor{T,nc}` or `ColorMixture{T,nc,rgbs}`). For instance, Landsat 8 data might look something like this:
+
+   ```julia
+   A = rand(0x0000:0x0fff, 11, 100, 100);
+   img = reinterpret(reshape, MultiChannelColor{N4f12,11}, A);
+   ```
+
+2. If the imported data have the color channel last, or use separate arrays for each channel, use the [StructArrays package](https://github.com/JuliaArrays/StructArrays.jl). For example:
+
+   ```julia
+   A = rand(0x0000:0x0fff, 100, 100, 11);
+   img = StructArray{MultiChannelColor{N4f12,11}}(A; dims=3)
+   ```
```

Comment:
If we provide a better tool in ImageCore in the spirit of https://github.com/JuliaImages/ImageCore.jl/pull/179, then this story might be changed.
Might be worth leaving a note about future improvement.

---

I was thinking of something like:

```julia
module ImageLayouts
const CHW = DenseImageLayout{:CHW}()
const CWH = DenseImageLayout{:CWH}()
const HWC = DenseImageLayout{:HWC}()
end
```

so that users can use a unified interface:

```julia
using ImageCore # which exports the ImageLayouts module

colorview(ImageLayouts.CHW, img)
```

we might end up with a new function, though -- pretty much like the `strel` concept I introduced in ImageMorphology.

Reply:
Yes, I really like the idea of these custom layouts!